src/handlers/payout/post.ts

```diff
@@ -84,160 +69,234 @@ export const incentivizeComments = async () => {
-    logger.info(`Incomplete payload. issue: ${issue}`);
-    return;
-  }
-
-  if (issue.state_reason !== StateReason.COMPLETED) {
-    logger.info("incentivizeCreatorComment: comment incentives skipped because the issue was not closed as completed");
-    return;
-  }
 
-  if (paymentPermitMaxPrice == 0 || !paymentPermitMaxPrice) {
-    logger.info(`incentivizeCreatorComment: skipping to generate permit2 url, reason: { paymentPermitMaxPrice: ${paymentPermitMaxPrice}}`);
-    return;
-  }
-
-  const issueDetailed = bountyInfo(issue);
+  const issueDetailed = bountyInfo(incentivesCalculation.issue);
   if (!issueDetailed.isBounty) {
     logger.info(`incentivizeCreatorComment: its not a bounty`);
-    return;
+    return { error: `incentivizeCreatorComment: its not a bounty` };
   }
 
-  const comments = await getAllIssueComments(issue.number);
+  const comments = await getAllIssueComments(incentivesCalculation.issue.number);
   const permitComments = comments.filter(
-    (content) => content.body.includes("Task Creator Reward") && content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot
+    (content) => content.body.includes(title) && content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot
   );
   if (permitComments.length > 0) {
     logger.info(`incentivizeCreatorComment: skip to generate a permit url because it has been already posted`);
-    return;
+    return { error: `incentivizeCreatorComment: skip to generate a permit url because it has been already posted` };
   }
 
-  const assignees = issue.assignees ?? [];
+  const assignees = incentivesCalculation.issue.assignees ?? [];
   const assignee = assignees.length > 0 ? assignees[0] : undefined;
   if (!assignee) {
     logger.info("incentivizeCreatorComment: skipping payment permit generation because `assignee` is `undefined`.");
-    return;
+    return { error: "incentivizeCreatorComment: skipping payment permit generation because `assignee` is `undefined`." };
   }
 
-  const description = await getIssueDescription(issue.number, "html");
+  const description = await getIssueDescription(incentivesCalculation.issue.number, "html");
   if (!description) {
     logger.info(`Skipping to generate a permit url because issue description is empty. description: ${description}`);
-    return;
+    return { error: `Skipping to generate a permit url because issue description is empty. description: ${description}` };
   }
   logger.info(`Getting the issue description done. description: ${description}`);
-  const creator = issue.user;
-  if (creator.type === UserType.Bot || creator.login === issue.assignee) {
+  const creator = incentivesCalculation.issue.user;
+  if (creator.type === UserType.Bot || creator.login === incentivesCalculation.issue.assignee) {
     logger.info("Issue creator assigneed himself or Bot created this issue.");
-    return;
+    return { error: "Issue creator assigneed himself or Bot created this issue." };
```

Comment:
Shouldn't these be two separate errors they seem unrelated

Reply:
Is issue creator not eligible for rewards if they are also the assignee? That doesn't make sense to me.